x-pack/solutions/observability/plugins/slo/public/components/slo/bulk_purge_confirmation_modal/bulk_purge_confirmation_modal.tsx

```diff
@@ -0,0 +1,166 @@
+  EuiDatePicker,
+  EuiFormRow,
+  EuiRadioGroup,
+  EuiSpacer,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { SLODefinitionResponse } from '@kbn/slo-schema';
+import React from 'react';
+import moment, { Moment } from 'moment';
+import { useBulkPurgeRollupData } from '../../../pages/slo_management/hooks/use_bulk_purge_rollup_data';
+
+export interface Props {
+  onCancel: () => void;
+  onConfirm: () => void;
+  items: SLODefinitionResponse[];
+}
+
+export function SloBulkPurgeConfirmationModal({ items, onCancel, onConfirm }: Props) {
+  const { mutateAsync: bulkPurge } = useBulkPurgeRollupData();
+
+  const [purgeDate, setPurgeDate] = React.useState<Moment | null>(moment());
+  const [purgeType, setPurgeType] = React.useState<string>('fixed_age');
+  const [forcePurge, setForgePurge] = React.useState<boolean>(false);
+  const [age, setAge] = React.useState('7d');
+
+  const basicCheckboxId = useGeneratedHtmlId({ prefix: 'basicCheckbox' });
+
+  return (
+    <EuiConfirmModal
+      buttonColor="danger"
+      data-test-subj="sloBulkPurgeConfirmationModal"
+      title={i18n.translate('xpack.slo.bulkPurgeConfirmationModal.title', {
+        defaultMessage: 'Purge {count} SLOs',
+        values: { count: items.length },
+      })}
+      cancelButtonText={i18n.translate('xpack.slo.bulkPurgeConfirmationModal.cancelButtonLabel', {
+        defaultMessage: 'Cancel',
+      })}
+      confirmButtonText={i18n.translate('xpack.slo.bulkPurgeConfirmationModal.confirmButtonLabel', {
+        defaultMessage: 'Purge',
+      })}
+      onCancel={onCancel}
+      onConfirm={async () => {
+        await bulkPurge({
+          list: items.map(({ id }) => id),
+          purgePolicy:
+            purgeType === 'fixed_age'
+              ? {
+                  purgeType: 'fixed_age',
+                  age,
+                }
+              : {
+                  purgeType: 'fixed_time',
+                  timestamp: purgeDate!.toISOString(),
+                },
+          force: forcePurge,
+        })
+          .then(() => {
+            onConfirm();
+          })
+          .catch((error) => {
+            console.error('error', error);
+          });
+      }}
+    >
+      {i18n.translate('xpack.slo.bulkPurgeConfirmationModal.descriptionText', {
+        defaultMessage:
+          'Rollup data for {count} SLOs will be purged according to the policy provided below.',
+        values: { count: items.length },
+      })}
+      <EuiSpacer size="m" />
+      <EuiFormRow
+        label={i18n.translate('xpack.slo.bulkPurgeConfirmationModal.purgeTypeLabel', {
+          defaultMessage: 'Purge type',
+        })}
+      >
+        <EuiRadioGroup
+          data-test-subj="sloBulkPurgeConfirmationModalPurgeType"
+          options={[
+            {
+              id: 'fixed_age',
+              label: i18n.translate('xpack.slo.bulkPurgeConfirmationModal.fixedAgeLabel', {
+                defaultMessage: 'Fixed age',
```

Comment:
I think we can be more descriptive for the user, something like "Purge data older than a specific duration" or something like that. wdyt?